### PR TITLE
add sign-in endpoint - Aisha Athman Lali

### PR DESCRIPTION
**Pull request by aishaathmanlali**
## Description

A PR about creating a /sign-in endpoint to validate user and password

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -263,4 +265,41 @@ router.post("/bid/:bidId/status", async (req, res) => {
...
+	const { email, password } = req.body;
+
+	if (!email || !password) {
+		return res.status(400).json({});
+	}
+
+	try {
+		const userResult = await db.query("SELECT * FROM users WHERE email = $1", [
+			email,
+		]);
```
Comment (soonick) [inline server/api.js:278]:
```suggestion
                hash  = bcrypt.hashSync(password, SALT);
		const userResult = await db.query("SELECT * FROM users WHERE email = $1 && password_hash = $2", [
			email, hash
		]);
```

Make SALT a global string with some random phrase. If there are no results, return `401`

File: server/api.js [hunk 2]
```diff
@@ -263,4 +265,41 @@ router.post("/bid/:bidId/status", async (req, res) => {
...
+		}
+
+		const isPasswordValid = await bcrypt.compare(password, user.password_hash);
+		if (!isPasswordValid) {
+			return res.status(401).json({});
+		}
+
+		const token = uuidv4();
+		const expirationDate = new Date();
+		expirationDate.setMonth(expirationDate.getMonth() + 1);
```
Comment (soonick) [inline server/api.js:294]: Does this work in december?

**Review by soonick — CHANGES_REQUESTED** [review]

File: server/api.js [hunk 3]
```diff
@@ -263,4 +266,43 @@ router.post("/bid/:bidId/status", async (req, res) => {
...
+
+		if (expirationDate.getDate() !== new Date().getDate()) {
+			expirationDate.setDate(0);
+		}
```
Comment (soonick) [inline server/api.js]: What is this?

**Review by soonick — CHANGES_REQUESTED** [review]

File: server/api.js (hunk 2 above)
Comment (soonick) [reply server/api.js:294]: I just verified it does

File: server/api.js (hunk 3 above)
Comment (aishaathmanlali) [reply server/api.js]: I have changed it back to the way it was before

**Review by soonick — APPROVED** [review]

**Pull request merged**